sdk/mixedreality/azure-mixedreality-remoterendering/src/test/java/com/azure/mixedreality/remoterendering/RemoteRenderingAsyncClientTest.java

```diff
@@ -0,0 +1,240 @@
+                assertTrue(conversion.getError().getMessage().toLowerCase(Locale.ROOT).contains("invalid input"));
+                assertTrue(conversion.getError().getMessage().toLowerCase(Locale.ROOT).contains("logs"));
+            })
+            .verifyComplete();
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getHttpClients")
+    public void sessionTest(HttpClient httpClient) {
+        RemoteRenderingAsyncClient client = getClient(httpClient);
+
+        BeginSessionOptions options = new BeginSessionOptions().setMaxLeaseTime(Duration.ofMinutes(4)).setSize(RenderingSessionSize.STANDARD);
+
+        String sessionId = getRandomId("ayncSessionTest");
+
+        PollerFlux<RenderingSession, RenderingSession> sessionPoller = client.beginSession(sessionId, options);
+
+        Flux<AsyncPollResponse<RenderingSession, RenderingSession>> terminalPoller = sessionPoller.map(response -> {
+            RenderingSession session = response.getValue();
+            assertEquals(sessionId, session.getId());
+            assertNotEquals(RenderingSessionStatus.ERROR, session.getStatus());
+            return response;
+        }).filter(response -> {
+            return ((response.getStatus() != LongRunningOperationStatus.NOT_STARTED)
+                && (response.getStatus() != LongRunningOperationStatus.IN_PROGRESS));
+        });
+
+        StepVerifier.create(terminalPoller)
+            .assertNext(response -> {
+                assertEquals(response.getStatus(), LongRunningOperationStatus.SUCCESSFULLY_COMPLETED);
+
+                RenderingSession readyRenderingSession = response.getValue();
+                assertEquals(readyRenderingSession.getStatus(), RenderingSessionStatus.READY);
+
+                assertTrue(readyRenderingSession.getMaxLeaseTime().toMinutes() == 4);
+                assertNotNull(readyRenderingSession.getHostname());
+                assertNotNull(readyRenderingSession.getArrInspectorPort());
+                assertNotNull(readyRenderingSession.getHostname());
+                assertEquals(readyRenderingSession.getSize(), options.getSize());
+            })
+            .verifyComplete();
+
+        StepVerifier.create(client.getSession(sessionId))
+            .assertNext(session -> {
+                assertEquals(session.getStatus(), RenderingSessionStatus.READY);
+                assertNotNull(session.getHostname());
+                assertNotNull(session.getArrInspectorPort());
+                assertNotNull(session.getHostname());
+                assertEquals(session.getSize(), options.getSize());
+            })
+            .verifyComplete();
+
+        UpdateSessionOptions updateOptions = new UpdateSessionOptions().maxLeaseTime(Duration.ofMinutes(5));
+
+        StepVerifier.create(client.updateSession(sessionId, updateOptions))
+            .assertNext(session -> {
+                assertTrue(session.getMaxLeaseTime().toMinutes() == 5);
```

Comment:
assertEquals(5, session.getMaxLeaseTime().toMinutes());
You can also define  method level variable `expectedLeaseTimeMinutes = 5 ` and use it here.